### PR TITLE
ready solution hw-26.1

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

File: src/main/java/com/internet/shop/controller/AddProductToStorageController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/product/addProduct.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controller/AddProductToStorageController.java:26]: What if we receive an empty input String?

File: src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private ShoppingCartService cartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = cartService.getByUserId(USER_ID);
+        long id = Long.parseLong(req.getParameter("id").replaceAll("\\D",""));
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java]: Explain, please, what are you replacing in here and why?
Also, add the same check 

_What if we receive an empty input String?_

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/user/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controller/RegistrationUserController.java]: Same check

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Let's fix some issues

File: src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java]: без replaceAll("\\D",""))...не парсило, викидало помилку. Причина була у файлі jsp
a href="${pageContext.request.contextPath}/products/delete?id=${product.id}&name=${product.name}&price=${product.price}">Delete</a>"
коли я прописую в одну стрічку попередню конструкцію то все норм і не треба replace робити

File: src/main/java/com/internet/shop/controller/AddProductToStorageController.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/AddProductToStorageController.java:26]: added validation of the form in jsp files

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/RegistrationUserController.java]: added validation of the form in jsp files

File: src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private ShoppingCartService cartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = cartService.getByUserId(USER_ID);
+        long id = Long.parseLong(req.getParameter("id"));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java:26]: can we just get product by id from storage the same way we did with shoppingCart and then call cartService.deleteProduct(shoppingCart, product) ?

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        if (password.equalsIgnoreCase(req.getParameter("repeatPwd"))) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/RegistrationUserController.java]: do we want password123 and PassWorD123 be the same passwords?

File: src/main/java/com/internet/shop/controller/GetAllProductsController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> products = productService.getAll();
+        req.setAttribute("products", products);
+        req.getRequestDispatcher("/WEB-INF/views/product/AllProducts.jsp").forward(req, resp);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/GetAllProductsController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/product/allProducts.jsp").forward(req, resp);
```

File: src/main/java/com/internet/shop/controller/DeleteUserController.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        long id = Long.parseLong(req.getParameter("id"));
+        userService.deleteById(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/DeleteUserController.java:23]: if we add shoppingCart by default in RegistrationController it makes sense to delete it here when we delete user

File: src/main/java/com/internet/shop/controller/ShoppingCartController.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/ShoppingCartController.java]:
```suggestion
public class GetShoppingCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import com.internet.shop.model.User;
+import com.internet.shop.service.ShoppingCartService;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationUserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/RegistrationUserController.java]:
```suggestion
public class UserRegistrationController extends HttpServlet {
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -8,9 +8,89 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>buyProducts</servlet-name>
+        <servlet-class>com.internet.shop.controller.AddProductToShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>buyProducts</servlet-name>
+        <url-pattern>/products/buy</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>shopping-cart/products/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -8,9 +8,89 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProducts</servlet-name>
+        <servlet-class>com.internet.shop.controller.DeleteProductFromShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProducts</servlet-name>
+        <url-pattern>/products/delete</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>shopping-cart/products/delete</url-pattern>
```

File: src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java (hunk 4 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java:26]: done

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java (hunk 5 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/RegistrationUserController.java]: done

File: src/main/java/com/internet/shop/controller/GetAllProductsController.java (hunk 6 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/GetAllProductsController.java]: done

File: src/main/java/com/internet/shop/controller/DeleteUserController.java (hunk 7 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/DeleteUserController.java:23]: done

File: src/main/java/com/internet/shop/controller/ShoppingCartController.java (hunk 8 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/ShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controller/RegistrationUserController.java (hunk 9 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/RegistrationUserController.java]: done

File: src/main/webapp/web.xml (hunk 10 above)
Comment (RomanHreits) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 11 above)
Comment (RomanHreits) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/shop/controller/DeleteUserController.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private ShoppingCartService cartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        long id = Long.parseLong(req.getParameter("id"));
+        userService.deleteById(id);
+        ShoppingCart byUserId = cartService.getByUserId(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/DeleteUserController.java]:
```suggestion
        ShoppingCart userCart = cartService.getByUserId(id);
```

File: src/main/webapp/WEB-INF/views/user/registration.jsp [hunk 13]
```diff
@@ -0,0 +1,40 @@
+<%--
+  Created by IntelliJ IDEA.
+  User: User
+  Date: 09.09.2020
+  Time: 12:51
+  To change this template use File | Settings | File Templates.
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/user/registration.jsp]: reomove autogenerated by idea comments

File: src/main/java/com/internet/shop/controller/DeleteUserController.java (hunk 12 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/DeleteUserController.java]: done

File: src/main/webapp/WEB-INF/views/user/registration.jsp (hunk 13 above)
Comment (RomanHreits) [reply src/main/webapp/WEB-INF/views/user/registration.jsp]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**